mira/modeling/amr/stockflow.py

```diff
@@ -0,0 +1,193 @@
+            model.template_model.annotations.name:
+            self.model_name = model.template_model.annotations.name
+        self.model_description = self.model_name
+        if model.template_model.annotations and \
+            model.template_model.annotations.description:
+            self.model_description = \
+                model.template_model.annotations.description
+
+        if model.template_model.time:
+            self.time = {'id': model.template_model.time.name}
+            if model.template_model.time.units:
+                self.time['units'] = {
+                    'expression': str(model.template_model.time.units.expression),
+                    'expression_mathml': expression_to_mathml(
+                        model.template_model.time.units.expression.args[0]),
+                }
+        else:
+            self.time = None
+
+        vmap = {}
+        for key, var in model.variables.items():
+            vmap[key] = name = var.concept.name or str(key)
+            display_name = var.concept.display_name or name
+            stocks_dict = {
+                'id': name,
+                'name': display_name,
+                'grounding': {
+                    'identifiers': {k: v for k, v in
+                                    var.concept.identifiers.items()
+                                    if k != 'biomodels.species'},
+                    'modifiers': var.concept.context,
+                },
+            }
+            if var.concept.units:
+                stocks_dict['units'] = {
+                    'expression': str(var.concept.units.expression),
+                    'expression_mathml': expression_to_mathml(
+                        var.concept.units.expression.args[0]
+                    ),
+                }
+
+            self.stocks.append(stocks_dict)
+            initial = var.data.get('expression')
+            if initial is not None:
+                initial_data = {
+                    'target': name,
+                    'expression': str(initial),
+                    'expression_mathml': expression_to_mathml(initial)
+                }
+                self.initials.append(initial_data)
+
+        for key, observable in model.observables.items():
+            display_name = observable.observable.display_name \
+                if observable.observable.display_name \
+                else observable.observable.name
+            obs_data = {
+                'id': observable.observable.name,
+                'name': display_name,
+                'expression': str(observable.observable.expression),
+                'expression_mathml': expression_to_mathml(
+                    observable.observable.expression.args[0]),
+            }
+            self.observables.append(obs_data)
+
+        for key, param in model.parameters.items():
+            if param.placeholder:
+                continue
+            param_dict = {'id': str(key)}
+            if key.startswith('p_'):
```

Comment:
An auxiliary should be introduced for _every_ parameter in the Model, and the naming convention for parameters that start with `p_` can be as below, but for all other parameters, we could introduce a convention to call the auxiliary key + '_aux'.